service/matching/taskQueueManager.go

```diff
@@ -630,3 +710,92 @@ func newIOContext() (context.Context, context.CancelFunc) {
 
 	return ctx, cancel
 }
+
+func (c *taskQueueManagerImpl) fetchMetadataFromRootPartitionOnInit(ctx context.Context) {
+	if c.metadataInitialFetch.Ready() {
+		return
+	}
+	_, err := c.fetchMetadataFromRootPartition(ctx)
+	c.metadataInitialFetch.Set(struct{}{}, err)
+}
+
+// fetchMetadataFromRootPartition fetches metadata from root partition iff this partition is not a root partition.
+// Returns the fetched data, if fetching was necessary and successful.
+func (c *taskQueueManagerImpl) fetchMetadataFromRootPartition(ctx context.Context) (*persistencespb.VersioningData, error) {
+	// Nothing to do if we are the root partition
+	// (for versioning - any later added metadata may need to not abort so early)
+	if c.taskQueueID.IsRoot() {
+		return nil, nil
+	}
+
+	curDat, err := c.db.GetVersioningData(ctx)
+	if err != nil && !errors.Is(err, errVersioningDataNotPresentOnPartition) {
+		return nil, err
+	}
+	curHash := HashVersioningData(curDat)
+
+	rootTqName := c.taskQueueID.GetRoot()
+	if len(curHash) == 0 {
+		// if we have no data, make sure we send a sigil value, so it's known we desire versioning data
+		curHash = []byte{0}
```

Comment:
why is empty curHash not enough to indicate we want latest data?

Reply:
No way to transmit it in the proto. The way (at least gogoproto works), if you pass an empty byte array that de-serializes to `nil`, same as if you didn't set the field at all. So something must be passed.

Reply:
but isn't nil convey the same info? we only skip if curHash match to current latest hash, otherwise we return latest data.

Reply:
No - if the requester sends nil for the field, it means they don't need/want that piece of data.